src/service/service.ts

```diff
@@ -283,6 +287,37 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
     });
   }
 
+  /**
+   * Broadcast TALKREQ message to all nodes in routing table
+   */
+  public async sendTalkReq(payload: Buffer, protocol: string): Promise<void> {
+    const msg = createTalkRequestMessage(payload, protocol);
+    for (const node of this.kadValues()) {
+      console.log(node.nodeId);
+      const sendStatus = this.sendRequest(node.nodeId, msg);
+      if (!sendStatus) {
+        log(`Failed to send TALKREQ message to node ${node.nodeId}`);
+      } else {
+        log(`Sent TALKREQ message to node ${node.nodeId}`);
+      }
+    }
+  }
+
+  /**
+   * Broadcast TALKRESP message to requestingi node
+   */
+  public async sendTalkResp(srcId: NodeId, request: ITalkReqMessage, payload: Buffer): Promise<void> {
+    const msg = createTalkResponseMessage(request, payload);
+    const enr = this.getKadValue(srcId);
+    const addr = await enr?.getFullMultiaddr("udp");
+    if (enr && addr) {
+      await this.sessionService.sendResponse(addr, srcId, msg);
+      log(`Sent TALKRESP message to node ${enr.id}`);
+    } else {
+      log(`Node ${srcId} not found`);
```

Comment:
I wonder if its worth splitting out 'node not found' from 'no ip + udp port found'